Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Booking.hs

```diff
@@ -84,3 +86,28 @@ data Booking = Booking
 
 data BookingType = SpecialZoneBooking | NormalBooking
   deriving (Show, Eq, Ord, Read, Generic, ToJSON, FromJSON, ToSchema)
+
+locationMappingMakerForBooking :: (MonadFlow m) => Booking -> m [DLocationMapping.LocationMapping]
```

Comment:
(MonadFlow m) is needed for those functions thats why could not call it inside create

Reply:
`create` does not work with `MonadFlow m`.. had tried earlier

Reply:
It should work, you should try it. @roman-bodavskiy Can you help her here?

Reply:
@umangj-28 you need `MonadIO` constraint because of `getPOSIXTime` function.
But we can replace it to our function `getCurrentTime`, then `MonadTime` constraint will be enough:

```
create :: Booking -> SqlDB ()
create dBooking = do
  mappings <- locationMappingMaker dBooking DLocationMapping.Booking
  Esq.runTransaction $
    withFullEntity dBooking $ \(booking, _, _, _fareParams) -> do
      Esq.create' booking
  QLocationMapping.createMany mappings


locationMappingMaker ::
  (MonadGuid m,
  MonadTime m,
  HasField "fromLocation" entity DLoc.Location,
  HasField "toLocation" entity [DLoc.Location],
  HasField "id" entity (Id entity)
  ) =>
  entity ->
  DLocationMapping.LocationMappingTags ->
  m [DLocationMapping.LocationMapping]
locationMappingMaker entity tag = do
  let searchWithIndexes = zip ([1 ..] :: [Int]) entity.toLocation
  toLocationMappers <- mapM (locationMappingInstanceMaker entity tag) searchWithIndexes
  fromLocationMapping <- locationMappingInstanceMaker entity tag (0, entity.fromLocation)
  return $ fromLocationMapping : toLocationMappers

locationMappingInstanceMaker ::
  (MonadGuid m,
  MonadTime m,
  HasField "id" entity (Id entity)
  ) =>
  entity ->
  DLocationMapping.LocationMappingTags ->
  (Int, DLoc.Location) ->
  m DLocationMapping.LocationMapping
locationMappingInstanceMaker entity tag location = do
  locationMappingId <- generateGUID
  now <- getCurrentTime
  let epochVersion = round $ utcTimeToPOSIXSeconds now
  let epochVersionLast5Digits = epochVersion `mod` 100000 :: Integer
  let locationMapping =
        DLocationMapping.LocationMapping
          { id = Id locationMappingId,
            tag,
            order = fst location,
            version = show epochVersionLast5Digits,
            tagId = entity.id.getId,
            location = snd location,
            ..
          }
  return locationMapping
```